mlkem/asm/aarch64/rej_uniform_asm_clean.S

```diff
@@ -284,15 +285,46 @@ table_data:
 bit_table_data:
     .short 0x1, 0x2, 0x4, 0x8, 0x10, 0x20, 0x40, 0x80
 
-.text
-.align 4
-.global     rej_uniform_asm_clean
-.global     _rej_uniform_asm_clean
-rej_uniform_asm_clean:
-_rej_uniform_asm_clean:
+/* Return multiple of 48: (n/48) * 48 */
+.macro round_48x    dst, src
+    mov     w8, #48
+    udiv    \dst, \src, w8
+    mul     \dst, \dst, w8
+.endm
+
+#define STACK_LOCS 2
+#define STACK_SIZE_TMP_OUTPUT (2*KYBER_N + 64) // We overwrite by max. 64 bytes in loop48
+#define STACK_SIZE_LOCS (8*STACK_LOCS)
+#define STACK_SIZE_GPRS (6 * 8)
+#define STACK_OFFSET_GPRS 0
+#define STACK_OFFSET_LOCS STACK_SIZE_GPRS
+#define STACK_OFFSET_TMP_OUTPUT (STACK_SIZE_LOCS + STACK_SIZE_GPRS)
+#define STACK_SIZE (STACK_SIZE_TMP_OUTPUT + STACK_SIZE_LOCS + STACK_SIZE_GPRS)
+
+#define STACK_OFFSET_RESULT (STACK_OFFSET_LOCS)
+
+.macro push_stack
+        sub sp, sp, #STACK_SIZE
+.endm
+
+.macro pop_stack
+        add sp, sp, #STACK_SIZE
+.endm
+
+.macro save_gprs
+    str     x8,       [sp, #(STACK_OFFSET_GPRS + 0)]
```

Comment:
@cothan It looks like we only need to stash `x30` -- `x8`, `x16` and `x17` don't have to be kep, and `x29` is not changed?

Reply:
I don't remember much, yes, if the x29 isn't changed then feel free to adjust the file as you see fit.

Reply:
Removed